sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/kafka/KafkaTableProviderIT.java

```diff
@@ -67,19 +67,25 @@
 import org.testcontainers.containers.KafkaContainer;
 
 /** This is an integration test for KafkaCSVTable. */
-public class KafkaCSVTableIT {
+public abstract class KafkaTableProviderIT {
   @Rule public transient TestPipeline pipeline = TestPipeline.create();
   @Rule public transient KafkaContainer kafka = new KafkaContainer();
 
-  private KafkaOptions kafkaOptions;
+  protected KafkaOptions kafkaOptions;
 
-  private static final Schema TEST_TABLE_SCHEMA =
+  protected static final Schema TEST_TABLE_SCHEMA =
       Schema.builder()
           .addNullableField("order_id", Schema.FieldType.INT32)
           .addNullableField("member_id", Schema.FieldType.INT32)
           .addNullableField("item_name", Schema.FieldType.INT32)
```

Comment:
Could you add more fields to this schema to exercise more types?

Reply:
CSV is quite problematic here as most fields (bool, Row, etc) are  indistinguishable by the default csv parser. I'll make a getSchema() abstract and provide different fields for Avro and Csv.